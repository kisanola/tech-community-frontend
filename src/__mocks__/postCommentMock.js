export default {
  author: { picture: 'img.jpg', username: 'username' },
  createdAt: '2019-10-07',
  body: 'Hello world',
};
