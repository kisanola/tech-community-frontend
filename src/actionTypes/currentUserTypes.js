export const SET_CURRENT_USER = 'SET_CURRENT_USER';
export const SET_IS_AUTH = 'SET_IS_AUTH';
