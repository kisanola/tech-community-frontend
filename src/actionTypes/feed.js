export const TOOGLE_LOADING = 'TOOGLE_LOADING';
export const SET_FEED = 'SET_FEED';
