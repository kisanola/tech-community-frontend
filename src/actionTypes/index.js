export const CHANGE_NAME = 'CHANGE_NAME';
export * from './currentUserTypes';
