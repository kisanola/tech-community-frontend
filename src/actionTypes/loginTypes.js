export const NO_ERROR = 'NO_ERROR';
export const LOGIN = 'LOGIN';
export const LOGIN_STARTED = 'LOGIN_STARTED';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGIN_INPUT_CHANGED = 'LOGIN_INPUT_CHANGED';
