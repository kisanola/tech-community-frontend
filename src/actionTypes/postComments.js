export const TOGGLE_LOADING = 'TOGGLE_LOADING';
export const SET_COMMENTS = 'SET_COMMENTS';
