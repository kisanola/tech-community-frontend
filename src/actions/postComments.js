import server from '../Api/server';
import * as types from '../actionTypes/postComments';

/**
 * Toggles the loading state
 * @return {object} action
 */
export const toggleLoading = (state) => ({
  type: types.TOGGLE_LOADING,
  payload: state,
});

/**
 * Set comments
 * @return {object} action
 */
export const setComments = (comments) => ({
  type: types.SET_COMMENTS,
  payload: comments,
});

/**
 * Get comments
 * @return {object} response
 */
export const getPostComments = (postSlug) => async (dispatch) => {
  dispatch(toggleLoading(true));

  try {
    const {
      data: { postComments },
    } = await server.get(`/posts/${postSlug}/Comments`);

    dispatch(setComments(postComments));
    dispatch(toggleLoading(false));
  } catch (e) {
    dispatch(toggleLoading(false));
  }
};
