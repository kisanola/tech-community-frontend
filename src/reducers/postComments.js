import * as types from '../actionTypes/postComments';

const reducer = (state = { loading: true }, { type, payload }) => {
  switch (type) {
    case types.TOGGLE_LOADING:
      return {
        ...state,
        loading: payload,
      };
    case types.SET_COMMENTS:
      return {
        ...state,
        items: payload,
      };
    default:
      return state;
  }
};

export default reducer;
