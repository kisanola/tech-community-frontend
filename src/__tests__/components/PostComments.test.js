import React from 'react';
import { shallow, mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import postCommentsReducer from '../../reducers/postComments';
import { TOGGLE_LOADING, SET_COMMENTS } from '../../actionTypes/postComments';
import {
  PostComments,
  mapStateToProps,
  mapDispatchToProps,
} from '../../components/PostComments/PostComments';
import postCommentMock from '../../__mocks__/postCommentMock';

describe('PostComments', () => {
  let wrapper;
  const props = {
    post: { likesCount: 2, slug: 'slug' },
    items: [postCommentMock],
    loading: false,
    onGetPostComments: jest.fn(),
  };

  describe('PostComments.jsx', () => {
    beforeEach(() => {
      wrapper = mount(
        <Router>
          <PostComments {...props} />
        </Router>,
      );
    });

    test('should render PostComments.jsx', () => {
      wrapper = shallow(<PostComments {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('State', () => {
    let mapped, dispatch;
    beforeEach(() => {
      dispatch = jest.fn();
      mapped = mapDispatchToProps(dispatch);
    });

    test('should return `mapStateToProps`', () => {
      const expectedState = {
        items: [postCommentMock],
        loading: false,
      };
      const state = mapStateToProps({
        postComments: { items: [postCommentMock], loading: false },
      });
      expect(state).toEqual(expectedState);
    });

    it('Should call `onGetPostComments`', () => {
      mapped.onGetPostComments('slug');
      expect(dispatch).toHaveBeenCalled();
    });
  });

  describe('PostComments reducer', () => {
    test('should test the reducer with type SET_COMMENTS', () => {
      const state = postCommentsReducer(
        { items: [] },
        { type: SET_COMMENTS, payload: [postCommentMock] },
      );
      expect(state.items).toEqual([postCommentMock]);
    });

    test('should test the reducer with type TOGGLE_LOADING', () => {
      const state = postCommentsReducer(
        {},
        { type: TOGGLE_LOADING, payload: true },
      );
      expect(state.loading).toBe(true);
    });
  });
});
