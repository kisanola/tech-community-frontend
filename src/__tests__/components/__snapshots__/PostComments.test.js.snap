// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostComments PostComments.jsx should render PostComments.jsx 1`] = `ShallowWrapper {}`;
